src/DefifaGovernor.sol

```diff
@@ -158,6 +181,28 @@ contract DefifaGovernor is
         return super.proposalThreshold();
     }
 
+    function verifyProofsAndExecute(
+        uint256 proposalId,
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash,
+        bytes32[] calldata _leaves
+    ) external {
+        bytes32 _root = proposalScorecardRoot[proposalId];
+        socrecardVerifier.verifyScorecard(_leaves, _root);
+        super.execute(targets, values, calldatas, descriptionHash);
```

Comment:
it is right `calldatas` contains the `setTierRedemptionWeights` data right so after proof is verified we call the execute method on governer.sol to execute the proposal 
did I interpret your comment wrong ?

Reply:
But couldn't you pass different scorecards in the `scorecards` and in the calldata for `setTierRedemptionWeights`? Its only the `scorecards` that are being checked, but those are not the ones that get send to `setTierRedemptionWeights`.

Reply:
the oz governer does check that 
- the `proposalID` is derived by the [hash function](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/Governor.sol#L133) so if the scorecards are modified then the proposal id would be diff.

- the `execute` method also calls this [state method](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/Governor.sol#L145) where the status of the proposal is derived & there is a [revert check as well regarding the same](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/Governor.sol#L298) 

so yeah it is prevented but we can add an extra check like we can call the `hashProposal` method on our end and verify with the proposal ID it would be costly but can be extra safety check let me know what you think